src/dataflow/internal/process/functions/source.ts

```diff
@@ -0,0 +1,68 @@
+import type {RArgument, RParseRequestProvider} from '../../../../r-bridge'
+import { sourcedDeterministicCountingIdGenerator} from '../../../../r-bridge'
+import {requestProviderFromFile} from '../../../../r-bridge'
+import {type NormalizedAst, type ParentInformation, removeTokenMapQuotationMarks, type RFunctionCall, RType} from '../../../../r-bridge'
+import {RShellExecutor} from '../../../../r-bridge/shell-executor'
+import {executeSingleSubStep} from '../../../../core'
+import {type DataflowProcessorInformation, processDataflowFor} from '../../../processor'
+import {type DataflowScopeName, type Identifier, overwriteEnvironments, type REnvironmentInformation, resolveByName} from '../../../environments'
+import type {DataflowInformation} from '../../info'
+import {dataflowLogger} from '../../../index'
+
+let sourceProvider = requestProviderFromFile()
+
+export function setSourceProvider(provider: RParseRequestProvider): void {
+	sourceProvider = provider
+}
+
+export function isSourceCall(name: Identifier, scope: DataflowScopeName, environments: REnvironmentInformation): boolean {
+	if(name != 'source')
```

Comment:
this is not correct. Theoretically one could alias source like this:

`a <- source` 

and now the import can happen like `a("other.R")`

Reply:
the check should happen for the resolved definition